### PR TITLE
Add tests for scaffold drawer and end drawer

### PR DESCRIPTION
**Pull request by ValentinVignal**
Contributes to https://github.com/flutter/flutter/issues/130459

It adds a test for
- `examples/api/lib/material/scaffold/scaffold.drawer.0.dart`
- `examples/api/lib/material/scaffold/scaffold.end_drawer.0.dart`

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Data Driven Fixes]: https://github.com/flutter/flutter/wiki/Data-driven-Fixes

### CODE REVIEW

File: examples/api/test/material/scaffold/scaffold.end_drawer.0_test.dart [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    // Opens the drawer by tapping the button at the center of the screen.
+    await tester.tap(find.widgetWithText(ElevatedButton, 'Open End Drawer'));
+    await tester.pumpAndSettle();
+
+    expect(find.byType(Drawer), findsOne);
+    expect(tester.getCenter(
+      find.byType(Drawer)).dx,
+      greaterThan(400), reason: 'The drawer should be on the right side of the screen',
+    );
```
Comment (bleroux) [inline examples/api/test/material/scaffold/scaffold.end_drawer.0_test.dart:25]:
```suggestion
    expect(
      tester.getCenter(find.byType(Drawer)).dx,
      greaterThan(400),
      reason: 'The drawer should be on the right side of the screen',
    );
```

File: examples/api/test/material/scaffold/scaffold.end_drawer.0_test.dart [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    await tester.tap(find.widgetWithText(ElevatedButton, 'Close Drawer'));
+    await tester.pumpAndSettle();
+
+    expect(find.byType(Drawer), findsNothing);
+
+    // Opens the drawer by tapping the drawer button in the app bar.
+    expect(tester.getCenter(
+      find.byType(EndDrawerButton)).dx,
+      greaterThan(400), reason: 'The drawer button should be on the right side of the app bar',
+    );
```
Comment (bleroux) [inline examples/api/test/material/scaffold/scaffold.end_drawer.0_test.dart:38]:
```suggestion
    expect(
      tester.getCenter(find.byType(EndDrawerButton)).dx,
      greaterThan(400),
      reason: 'The drawer button should be on the right side of the app bar',
    );
```

File: examples/api/test/material/scaffold/scaffold.drawer.0_test.dart [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+    // Opens the drawer by tapping the button at the center of the screen.
+    await tester.tap(find.widgetWithText(ElevatedButton, 'Open Drawer'));
+    await tester.pumpAndSettle();
+
+    expect(find.byType(Drawer), findsOne);
+    expect(tester.getCenter(
+      find.byType(Drawer)).dx,
+      lessThan(400), reason: 'The drawer should be on the left side of the screen',
+    );
```
Comment (bleroux) [inline examples/api/test/material/scaffold/scaffold.drawer.0_test.dart:25]:
```suggestion
    expect(
      tester.getCenter(find.byType(Drawer)).dx,
      lessThan(400),
      reason: 'The drawer should be on the left side of the screen',
    );
```

File: examples/api/test/material/scaffold/scaffold.drawer.0_test.dart [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    await tester.tap(find.widgetWithText(ElevatedButton, 'Close Drawer'));
+    await tester.pumpAndSettle();
+
+    expect(find.byType(Drawer), findsNothing);
+
+    // Opens the drawer by tapping the drawer button in the app bar.
+    expect(tester.getCenter(
+      find.byType(DrawerButton)).dx,
+      lessThan(400), reason: 'The drawer button should be on the left side of the app bar',
+    );
```
Comment (bleroux) [inline examples/api/test/material/scaffold/scaffold.drawer.0_test.dart:38]:
```suggestion
    expect(
      tester.getCenter(find.byType(DrawerButton)).dx,
      lessThan(400),
      reason: 'The drawer button should be on the left side of the app bar',
    );
```

File: examples/api/test/material/scaffold/scaffold.drawer.0_test.dart [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    await tester.tap(find.byType(DrawerButton));
+    await tester.pumpAndSettle();
+
+    expect(find.byType(Drawer), findsOne);
+    expect(find.text('This is the Drawer'), findsOne);
+
+    // Closes the drawer by tapping outside the drawer
```
Comment (bleroux) [inline examples/api/test/material/scaffold/scaffold.drawer.0_test.dart:45]:
```suggestion
    // Closes the drawer by tapping outside the drawer.
```

File: examples/api/test/material/scaffold/scaffold.end_drawer.0_test.dart [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    await tester.tap(find.byType(EndDrawerButton));
+    await tester.pumpAndSettle();
+
+    expect(find.byType(Drawer), findsOne);
+    expect(find.text('This is the Drawer'), findsOne);
+
+    // Closes the drawer by tapping outside the drawer
```
Comment (bleroux) [inline examples/api/test/material/scaffold/scaffold.end_drawer.0_test.dart:45]:
```suggestion
    // Closes the drawer by tapping outside the drawer.
```

**Review by bleroux — APPROVED** [review]
LGTM with some formatting nits to correct.
Thanks for the contribution!

File: examples/api/test/material/scaffold/scaffold.drawer.0_test.dart [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+    expect(find.byType(Drawer), findsOne);
+    expect(tester.getCenter(
+      find.byType(Drawer)).dx,
+      lessThan(400),
+      reason: 'The drawer should be on the left side of the screen',
+    );
+    expect(find.text('This is the Drawer'), findsOne);
+
+    // Closes the drawer by tapping the button inside the drawer.
```
Comment (TahaTesser) [inline examples/api/test/material/scaffold/scaffold.drawer.0_test.dart:29]:
```suggestion
    // Close the drawer by tapping the button inside the drawer.
```

File: examples/api/test/material/scaffold/scaffold.drawer.0_test.dart [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    await tester.tap(find.widgetWithText(ElevatedButton, 'Close Drawer'));
+    await tester.pumpAndSettle();
+
+    expect(find.byType(Drawer), findsNothing);
+
+    // Opens the drawer by tapping the drawer button in the app bar.
```
Comment (TahaTesser) [inline examples/api/test/material/scaffold/scaffold.drawer.0_test.dart:35]:
```suggestion
    // Open the drawer by tapping the drawer button in the app bar.
```

File: examples/api/test/material/scaffold/scaffold.drawer.0_test.dart [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+      lessThan(400),
+      reason: 'The drawer button should be on the left side of the app bar',
+    );
+    await tester.tap(find.byType(DrawerButton));
+    await tester.pumpAndSettle();
+
+    expect(find.byType(Drawer), findsOne);
+    expect(find.text('This is the Drawer'), findsOne);
+
+    // Closes the drawer by tapping outside the drawer.
```
Comment (TahaTesser) [inline examples/api/test/material/scaffold/scaffold.drawer.0_test.dart:47]:
```suggestion
    // Close the drawer by tapping outside the drawer.
```

File: examples/api/test/material/scaffold/scaffold.end_drawer.0_test.dart [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+
+void main() {
+  testWidgets('The page should contain an end drawer than can be opened and closed', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const example.EndDrawerExampleApp(),
+    );
+
+    expect(find.byType(Drawer), findsNothing);
+
+    // Opens the drawer by tapping the button at the center of the screen.
```
Comment (TahaTesser) [inline examples/api/test/material/scaffold/scaffold.end_drawer.0_test.dart:17]:
```suggestion
    // Open the drawer by tapping the button at the center of the screen.
```

File: examples/api/test/material/scaffold/scaffold.end_drawer.0_test.dart [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+
+    expect(find.byType(Drawer), findsOne);
+    expect(tester.getCenter(
+      find.byType(Drawer)).dx,
+      greaterThan(400),
+      reason: 'The drawer should be on the right side of the screen',
+    );
+    expect(find.text('This is the Drawer'), findsOne);
+
+    // Closes the drawer by tapping the button inside the drawer.
```
Comment (TahaTesser) [inline examples/api/test/material/scaffold/scaffold.end_drawer.0_test.dart:29]:
```suggestion
    // Close the drawer by tapping the button inside the drawer.
```

File: examples/api/test/material/scaffold/scaffold.end_drawer.0_test.dart [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+    await tester.tap(find.widgetWithText(ElevatedButton, 'Close Drawer'));
+    await tester.pumpAndSettle();
+
+    expect(find.byType(Drawer), findsNothing);
+
+    // Opens the drawer by tapping the drawer button in the app bar.
```
Comment (TahaTesser) [inline examples/api/test/material/scaffold/scaffold.end_drawer.0_test.dart:35]:
```suggestion
    // Open the drawer by tapping the drawer button in the app bar.
```

File: examples/api/test/material/scaffold/scaffold.end_drawer.0_test.dart [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+      greaterThan(400),
+      reason: 'The drawer button should be on the right side of the app bar',
+    );
+    await tester.tap(find.byType(EndDrawerButton));
+    await tester.pumpAndSettle();
+
+    expect(find.byType(Drawer), findsOne);
+    expect(find.text('This is the Drawer'), findsOne);
+
+    // Closes the drawer by tapping outside the drawer.
```
Comment (TahaTesser) [inline examples/api/test/material/scaffold/scaffold.end_drawer.0_test.dart:47]:
```suggestion
    // Close the drawer by tapping outside the drawer.
```

**Review by TahaTesser — CHANGES_REQUESTED** [review]

**Review by TahaTesser — APPROVED** [review]
LGTM

**Comment by bleroux** [comment]
@ValentinVignal This was reverted due to an infra failure.
Nothing wrong with your PR, but there is a bad infra failure since last week.
Thank you to file this PR again and, before landing it, we will wait before for https://github.com/flutter/flutter/issues/149851 to be fixed.

**Comment by ValentinVignal** [comment]
Sure, I'll do that

**Comment by ValentinVignal** [comment]
Here it is @bleroux  https://github.com/flutter/flutter/pull/150047

**Pull request merged**